test/e2e/kubectl.go

```diff
@@ -1695,6 +1695,59 @@ metadata:
 			}
 		})
 	})
+
+	framework.KubeDescribe("Kubectl deployment reaper", func() {
+		var nsFlag string
+		var name string
+		var labelFlag string
+
+		BeforeEach(func() {
+			nsFlag = fmt.Sprintf("--namespace=%v", ns)
+			name = "e2e-test-nginx-deployment"
+			labelFlag = fmt.Sprintf("--labels=run=%v", name)
+		})
+
+		It("should delete deployment and all of the resources managed by it", func() {
+			// Create deployment with a specific label so that replicasets and pods also have the label
+			By("running the image " + nginxImage)
+			framework.RunKubectlOrDie("run", name, "--image="+nginxImage, nsFlag, labelFlag)
+
+			// If the pod is created successfully, it means deployment and replicasets are created successfully too
+			By("verifying the pod controlled by " + name + " gets created")
+			label := labels.SelectorFromSet(labels.Set(map[string]string{"run": name}))
+			err := testutils.WaitForPodsWithLabelRunning(c, ns, label)
+			if err != nil {
+				framework.Failf("Failed waiting for pod controlled by %s to be running: %v", name, err)
+			}
+
+			By("deleting the deployment " + name)
+			framework.RunKubectlOrDie("delete", "deployment", name, nsFlag, "--cascade=true")
+
+			By("verifying deployment was deleted")
+			err = wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+				_, err := c.ExtensionsV1beta1().Deployments(ns).Get(name, metav1.GetOptions{})
```

Comment:
Referring [this](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/stop.go#L90), deployments work for both `AppsV1beta1` and `ExtensionsV1beta1`.

Do we prefer `AppsV1beta1` over `ExtensionsV1beta1`?